### PR TITLE
Create integration environments

### LINKED ISSUE #5222: Create redistributable boards for Splunk
## Description

We want to implement new boards and panels based on our documentation, which will cover the following:

- [Security analysis](https://documentation.wazuh.com/current/_images/screenshot-01-Wazuh-Security-Analytics1.png)
- [Intrusion detection](https://documentation.wazuh.com/current/_images/screenshot-02-Wazuh-Intrusion-Detection1.png)
- [Log data analysis](https://documentation.wazuh.com/current/_images/screenshot-03-Wazuh-Log-Data-Analysis1.png)
- [File integrity monitoring](https://documentation.wazuh.com/current/_images/screenshot-04-Wazuh-File-Integrity-Monitoring1.png)
- [Vulnerability detection](https://documentation.wazuh.com/current/_images/screenshot-05-Wazuh-Vulnerability-Detection1.png)
- [Configuration assessment](https://documentation.wazuh.com/current/_images/screenshot-06-Wazuh-Configuration-Assessment1.png)
- [Incident response](https://documentation.wazuh.com/current/_images/screenshot-07-Wazuh-Incident-Response1.png)
- [Regulatory compliance](https://documentation.wazuh.com/current/_images/screenshot-08-Wazuh-Regulatory-Compliance1.png)
- [Cloud security](https://documentation.wazuh.com/current/_images/screenshot-09-Wazuh-Cloud-Security1.png)
- [Containers security](https://documentation.wazuh.com/current/_images/screenshot-10-Wazuh-Containers-Security1.png)

We want to use our sample data as the base for its creation. 

- Create a Splunk environment using the latest version
- Upload our sample data collection
- Using Splunk dashboard studio, create each query and dashboard to compose each of the dashboards

Tracked in: 
- https://github.com/wazuh/wazuh-kibana-app/issues/5182

### PR DESCRIPTION
**Pull request by Tostti**
### Description

Environment and dashboards for Splunk, Elastic, and Opensearch
 
### Issues Resolved

#5222 #5185 #5402 

### Evidence

Captura de pantalla 2023-04-24 a la(s) 10 38 13

### Elastic

Captura de pantalla 2023-04-24 a la(s) 15 37 56

### Opensearch

Captura de pantalla 2023-04-24 a la(s) 12 50 03

### Splunk

Captura de pantalla 2023-04-24 a la(s) 13 53 04

### Splunk

The Splunk dashboards are located in `extra/dashboards/Splunk`. The steps to import them to the indexer are the following:

- Open a dashboard file and copy all its content
- In the indexer navigate to `Search & Reporting`, `Dashboards`, click `Create New Dashboard`, write the title and select `Dashboard Studio`, select `Grid` and click on `Create`
- On the top menu, there is a `Source` icon. Click on it, and replace all the content with the copied content from the dashboard file. After that, click on `Back` and click on `Save`.
- Repeat the steps for all the desired dashboards.

### Elastic

The Elastic dashboards are located in `docker/integrations/extra/dashboards/elastic`. The steps to import them to the indexer are the following:

- Open the Elastic web interface
- Expand the left bar, and go to `Stack management`
- Click on `Saved Objects`, select `Import`, click on the `Import` icon and browse the dashboard file. It is possible to import only the desired dashboard, or the file `all-dashboards.ndjson`, that contains all the dashboards.
- Click on Import.
- Repeat the steps for all the desired dashboards.

After that, the dashboard should be imported. It can be seen opening the left bar and selecting `Dashboard`.

### Opensearch

The Elastic dashboards are located in `docker/integrations/extra/dashboards/opensearch`. The steps to import them to the indexer are the following:

- Open the Opensearch web interface
- Expand the left bar, and go to `Stack management`
- Click on `Saved Objects`, select `Import`, click on the `Import` icon and browse the dashboard file. It is possible to import only the desired dashboard, or the file `all-dashboards.ndjson`, that contains all the dashboards.
- Click on Import.
- Repeat the steps for all the desired dashboards.

After that, the dashboard should be imported. It can be seen opening the left bar and selecting `Dashboard`.

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by yenienserrano** [comment]
Kibana 8.6.2:
<details><summary>Screenshots</summary>

image

image

image

</details> 

LGTM

**Comment by yenienserrano** [comment]
Splunk 9.0.4:

<details><summary>Details</summary>

image

image

image

</details> 

LGTM

**Comment by chantal-kelm** [comment]
Kibana 8.6.2:
<details><summary>Screenshots</summary>

image

Splunk 9.0.4:

<details><summary>Details</summary>

image

**Comment by yenienserrano** [comment]
Do all versions have the same unchanged files?

If so, I think it would be good to leave it only once so as not to repeat code and have to keep several codes the same.

image

**Review by Machi3mfl — APPROVED** [review]

**Comment by yenienserrano** [comment]
# Tests evidence

## Splunk

<details>
<summary>Installation</summary>



</details>

<details> 
<summary>Dashboards</summary>




## Kibana 🟢 


<details>
<summary>Installation</summary>

image

</details>

<details>
<summary>Dashboards</summary>

Vulnerabilities

image

Security Events

image

PCI DSS

image

Malware detection

image

Incident response

image

Docker listener 

image

Amazon aws

image


</details>


## OpenSearch

<details>
<summary>Installation</summary>



</details>

<details>
<summary>Dashboards</summary>




</details>

**Comment by yenienserrano** [comment]
It would be good to have the same time filter for all dashboards.

The only one that has policy monitoring is splunk?

**Review by yenienserrano — APPROVED** [review]
LGTM

**Comment by chantal-kelm** [comment]
> It would be good to have the same time filter for all dashboards.
> 
> The only one that has policy monitoring is splunk?

Corrected the time filter, the dashboard policy-monitoring had the wrong name, corrected to security-events.

File: docker/integrations/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+- Internet connection
+- Docker
+- Docker compose
+
+## Usage
+
+In the .env file it is possible to configure the desired version of the Wazuh stack. It will only work with already released versions.
+
+After that, running `docker compose up -d` will start all the containers. Once the start process finishes, the integrations will be configured. It is necessary to manually start the Splunk integration from manager by running `/opt/splunkforwarder/bin/splunk start --accept-license` in the Wazuh manager container.To stop the environment and cleare it, use `docker compose down`.
```
Comment (asteriscos) [inline docker/integrations/README.md:29]: There's a typo in "cleare".

**Review by asteriscos — COMMENTED** [review]
CR: :orange_circle:  There's a typo in the readme.
Test: :heavy_check_mark: 

# Tests evidence

## Splunk :green_circle: 

<details>

<summary>Dashboards</summary>

Screenshot from 2023-04-25 18-36-00


</details>

## Kibana 🟢 

<details>
<summary>Dashboards</summary>

Screenshot from 2023-04-25 18-47-11


</details>


## OpenSearch 🟢 

<details>
<summary>Dashboards</summary>

Screenshot from 2023-04-25 19-43-35


</details>

File: docker/integrations/README.md (hunk 1 above)
Comment (chantal-kelm) [reply docker/integrations/README.md]: Fixed [7f77da9](https://github.com/wazuh/wazuh-kibana-app/pull/5308/commits/7f77da90dc8c3e59562e58cdb6adf825c10f2e7d)

**Review by asteriscos — APPROVED** [review]
LGTM :heavy_check_mark:

**Review by Machi3mfl — APPROVED** [review]

**Pull request merged**